tests/widgets/test_colour_text.py

```diff
@@ -0,0 +1,110 @@
+
+import unittest
+from urwid import AttrSpec
+from unittest import mock
+
+from alot.settings.const import settings
+from alot.settings import theme
+from alot.widgets.colour_text import parse_quotes
+
+
+DUMMY_THEME = """\
+[bufferlist]
+    line = '', '', '', '', '', ''
+    line_even = '', '', '', '', '', ''
+    line_focus = '', '', '', '', '', ''
+    line_odd = '', '', '', '', '', ''
+[envelope]
+    body = '', '', '', '', '', ''
+    header = '', '', '', '', '', ''
+    header_key = '', '', '', '', '', ''
+    header_value = '', '', '', '', '', ''
+[global]
+    body = '', '', '', '', '', ''
+    footer = '', '', '', '', '', ''
+    notify_error = '', '', '', '', '', ''
+    notify_normal = '', '', '', '', '', ''
+    prompt = '', '', '', '', '', ''
+    tag = '', '', '', '', '', ''
+    tag_focus = '', '', '', '', '', ''
+[help]
+    section = '', '', '', '', '', ''
+    text = '', '', '', '', '', ''
+    title = '', '', '', '', '', ''
+[taglist]
+    line_even = '', '', '', '', '', ''
+    line_focus = '', '', '', '', '', ''
+    line_odd = '', '', '', '', '', ''
+[namedqueries]
+    line_even = '', '', '', '', '', ''
+    line_focus = '', '', '', '', '', ''
+    line_odd = '', '', '', '', '', ''
+[search]
+    focus = '', '', '', '', '', ''
+    normal = '', '', '', '', '', ''
+    [[threadline]]
+        focus = '', '', '', '', '', ''
+        normal = '', '', '', '', '', ''
+[thread]
+    quote_level_2 = '', '', '', '', '#880', '#886'
+    quote_level_3 = '', '', '', '', '#880', '#886'
+    quote_level_4 = '', '', '', '', '#580', '#586'
+    quote_level_5 = '', '', '', '', '#880', '#886'
+    quote_level_6 = '', '', '', '', '#280', '#286'
+    quote_level_7 = '', '', '', '', '#180', '#186'
+    arrow_bars = '', '', '', '', '', ''
+    arrow_heads = '', '', '', '', '', ''
+    attachment = '', '', '', '', '', ''
+    attachment_focus = '', '', '', '', '', ''
+    body = '', '', '', '', 'h102', 'h103'
+    header = '', '', '', '', '', ''
+    header_key = '', '', '', '', '', ''
+    header_value = '', '', '', '', '', ''
+    [[summary]]
+        even = '', '', '', '', '', ''
+        focus = '', '', '', '', '', ''
+        odd = '', '', '', '', '', ''
+"""
+
+SETTINGS = {'quote_symbol': '>',
+            'colourmode': 256
+            }
+
+SETTINGS_test_symbol_regex = {'quote_symbol': '([Aa]|<>.b)',
+                              'colourmode': 256
+                              }
+
+
+class TestQuoteLevel(unittest.TestCase):
+
+    @mock.patch(
+        'alot.commands.globals.settings._config', SETTINGS)
+    @mock.patch(
+        'alot.commands.globals.settings._theme',
+        theme.Theme(DUMMY_THEME.splitlines()))
+    def test_quote_level_get_equals_corresponding_attribute(self):
+        self.assertEqual(settings.get('quote_symbol'), '>')
+
+        self.assertIsNone(parse_quotes(''))
+        self.assertIsNone(parse_quotes('test no quote'))
+
+        # Undefined value
+        self.assertIsNone(parse_quotes('>'))
```

Comment:
Why is this None?  I was expecting it to be the color for level 1. But that is not defined. So maybe this is the default value from the spec?  In this case I would make one extra test method that tests default values.

Reply:
When an undefined `quote_level` theming attribute is requested but not found, `'default'` is returned by `settings.get_theming_attribute`.

The return value from `get_quote_colour` is either `None` if no quote were found, `'default'` if a quote was found but the theming was undefined or an `AttrSpec` instance if this is a quote but its theming value is defined.

In the first two cases, `parse_quotes` will return `None` and this is wanted.   
>  In this case I would make one extra test method that tests default values.

I added on such test case.